src/mm.rs

```diff
@@ -73,20 +76,41 @@ pub fn load_user_app(app_name: &str) -> AxResult<(VirtAddr, VirtAddr, AddrSpace)
         // TDOO: flush the I-cache
     }
 
-    let ustack_top = uspace.end();
-    let ustack_vaddr = ustack_top - crate::USER_STACK_SIZE;
+    let ustack_base = uspace.end();
+    let ustack_size = crate::USER_STACK_SIZE;
+    let ustack_top = ustack_base - ustack_size;
     debug!(
         "Mapping user stack: {:#x?} -> {:#x?}",
-        ustack_vaddr, ustack_top
+        ustack_top, ustack_base
     );
+    // FIXME: Add more arguments and environment variables
+    let args = vec![app_name.to_string()];
+    let (stack_data, ustack_bottom) =
+        elf_parser::get_app_stack_region(args, &Vec::new(), elf_info.auxv, ustack_top, ustack_size);
     uspace.map_alloc(
-        ustack_vaddr,
-        crate::USER_STACK_SIZE,
+        ustack_top,
+        ustack_size,
         MappingFlags::READ | MappingFlags::WRITE | MappingFlags::USER,
-        false,
+        true,
     )?;
-    info!("New user address space: {:#x?}", uspace);
-    Ok((elf_info.entry, ustack_top, uspace))
+
+    for (idx, vaddr) in memory_addr::PageIter4K::new(ustack_top, ustack_base)
```

Comment:
Why we need to iterate the whole stack? `env` and `aux` take up only a small portion of the user stack.